### ISSUE TITLE
Add seed parameter to a Dense layer so that kernel initializers could use it for reproducibility

### ISSUE THREAD

**Issue by deil87** [tensorflow/tensorflow#38974]
**System information**
- TensorFlow version (you are using): tf.version.GIT_VERSION = v2.1.0-rc2-17-ge5bf8de410
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Right now `initializers.py` package has initializers that could be seeded but there is no way for users to specify seed when model and its layers are being specified.

**Will this change the current api? How?**
Yes, adds seed parameter to a Dense layer. So docs would probably need to be updated.

**Who will benefit with this feature?**
Anyone who will want to have reproducible results from their neural network. Some initializers already have `seed` parameter in their API so this PR is just an attempt to make integration of APIs more complete.
**Any Other info.**

- As there probably are other Layers that have kernel initializers we need to check them and add support for  them as well
- regularizers and constraints theoretically can also leverage availability of seed but in current implementation I don't see any of them to be using seeds
- I was trying to run all tests in tensorflow by using Docker but not sure that I'm doing it correctly. It was taking too long and I could see C++ tests to be running. I would appreciate some guidance here. So I was able to run only my own test that I will submit within upcoming PR

**Comment by ravikyram** [comment]
Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by deil87** [comment]
@ravikyram here is a PR https://github.com/tensorflow/tensorflow/pull/38975 .  During benchmarking you might want to fix everything in your NN and see effects from changes in a particular part of the model. This helps during testing and research.

**Comment by deil87** [comment]
@ravikyram people out there are trying to do this: https://machinelearningmastery.com/reproducible-results-neural-networks-keras/

extract from this article

>What if I Am Still Getting Different Results?
To re-iterate, the most robust way to report results and compare models is to repeat your experiment many times (30+) and use summary statistics.
If this is not possible, you can get 100% repeatable results by seeding the random number generators used by your code. The solutions above should cover most situations, but not all.
What if you have followed the above instructions and still get different results from the same algorithm on the same data?
It is possible that there are other sources of randomness that you have not accounted for.

So it means people we go to check other libraries for randomness but there is still randomness in tensorflow itself that users don't have easy control over

**Comment by qlzh727** [comment]
If you want to have an initializer that is reproducible, you can create the initializer and pass it to the dense layer, as the param for kernel_initializer or bias_initalizer. I think the current API is flexible enough without adding the seed param. I am closing this issue for now. Let me know if this is any other issue.

Eg:
```
init = tf.keras.initializers.GlorotNormal(seed=123)
dense = tf.keras.layers.Dense(4, kernel_initializer=init)
```
